mempool/mempool.go

```diff
@@ -67,11 +67,11 @@ type (
 // It allows customization of the underlying mempools, verification functions,
 // and broadcasting functions used by the sdkmempool.
 type EVMMempoolConfig struct {
-	TxPool        *txpool.TxPool
-	CosmosPool    sdkmempool.ExtMempool
-	AnteHandler   sdk.AnteHandler
-	BroadCastTxFn func(txs []*ethtypes.Transaction) error
-	BlockGasLimit uint64 // Block gas limit from consensus parameters
+	LegacyPoolConfig *legacypool.Config
+	CosmosPoolConfig *sdkmempool.PriorityNonceMempoolConfig[math.Int]
```

Comment:
I kept this as a math.Int because I didn't feel like making this generic would be entirely useful since we usually rely on gas/fees for priorities. Could be wrong here though.